cogs/stockmarket/utils.py

```diff
@@ -0,0 +1,180 @@
+from math import ceil
+import random
+import time
+import datetime
+import matplotlib.pyplot as plt
+import yfinance as yf  # pip install yfinance
+import discord
+from datetime import datetime
+
+from redbot.core import bank, commands, Config
+
+from .constants import *
+
+
+class market_utils:
+    def __init__(self):
+        self.config = Config.get_conf(self, identifier=835888251, force_registration=True)
+
+    async def is_invalid_ticker(self, ticker_info: dict):
+        if ticker_info["regularMarketPrice"] is None:
+            return True
+        if ticker_info["quoteType"] == "INDEX":
+            return True
+        if ticker_info["quoteType"] == "MUTUAL FUND":
+            return True
+
+        return False
+
+    async def cancel_transaction(self, sent_message: discord.Message):
+        new_embed = discord.Embed(
+            color=0xFF0000, title=BROKER_NAME, description=TRANSACTION_CANCELLED_STR
+        )
+        await sent_message.edit(embed=new_embed)
+
+    async def is_event_claimable(
+        self, current_unix: float, most_recent_event_unix: float, initial_DOP: float
+    ):
+        return current_unix < most_recent_event_unix < initial_DOP
+
+    async def yf_download(self, ticker: str, period_str: str, interval_str: str):
+        return yf.download(ticker, period=period_str, interval=interval_str)
+
+    async def yf_ticker_info(self, ticker: str):
+        return yf.Ticker(ticker).info
+
+    async def yf_ticker_actions(self, ticker: str):
+        return yf.Ticker(ticker).actions
+
+    async def get_embed_chart(self, ticker: str, period_str: str = "1d", interval_str: str = "5m"):
+        """Gets the chart of the requested stock"""
+        plt.clf()
+        ticker = ticker.upper()
+        # Get the data.. takes like 0.2s
+        data = await self.yf_download(ticker, period_str, interval_str)
+
+        if len(data["Close"]) == 0:
+            return False
+
+        # Plot the data using the close data
+        data["Close"].plot(color="red" if (data["Close"][0] - data["Close"][-1] > 0) else "green")
+
+        plt.title(ticker)
+        plt.ylabel("Price")
+        plt.grid()
+
+        # convert to webp????
+        plt.savefig(EMBED_LOCATION, bbox_inches="tight")  # note: saved as .jpg
+        return True
+
+    # so messy...
+    async def build_embed(self, ctx: commands.Context, price: int, ticker: str, ticker_info: dict):
+        """Builds the embed"""

```

Comment:
nit: does this need to be inside the class? I don't see `self` being used anywhere in this helper.